x-pack/platform/plugins/shared/spaces/public/space_selector/components/space_table.tsx

```diff
@@ -0,0 +1,205 @@
+ * 2.0.
+ */
+
+import {
+  EuiBasicTable,
+  type EuiBasicTableColumn,
+  EuiLink,
+  EuiLoadingSpinner,
+  EuiText,
+  EuiTextColor,
+} from '@elastic/eui';
+import React, { Component, lazy, Suspense } from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import type { Space } from '../../../common';
+import { addSpaceIdToPath, ENTER_SPACE_PATH } from '../../../common';
+import { getSpaceAvatarComponent } from '../../space_avatar';
+import { SpaceSolutionBadge } from '../../space_solution_badge';
+
+// Lazy load the space avatar component
+const LazySpaceAvatar = lazy(() =>
+  getSpaceAvatarComponent().then((component) => ({ default: component }))
+);
+
+interface Props {
+  spaces: Space[];
+  serverBasePath: string;
+  loading?: boolean;
+}
+
+interface State {
+  pageIndex: number;
+  pageSize: number;
+  sortField?: keyof Space;
+  sortDirection?: 'asc' | 'desc';
+}
+
+export class SpaceTable extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      pageIndex: 0,
+      pageSize: 10,
+      sortField: 'name',
+      sortDirection: 'asc',
+    };
+  }
+
+  private onTableChange = ({ page, sort }: any) => {
+    if (page) {
+      this.setState({
+        pageIndex: page.index,
+        pageSize: page.size,
+      });
+    }
+
+    if (sort) {
+      this.setState({
+        sortField: sort.field,
+        sortDirection: sort.direction,
+      });
+    }
+  };
+
+  private renderSpaceAvatar = (space: Space) => {
+    return (
+      <Suspense fallback={<EuiLoadingSpinner size="s" />}>
+        <LazySpaceAvatar space={space} size="s" announceSpaceName={false} />
+      </Suspense>
+    );
+  };
+
+  private renderSpaceName = (space: Space) => {
+    const spaceUrl = addSpaceIdToPath(this.props.serverBasePath, space.id, ENTER_SPACE_PATH);
+
+    return (
+      <EuiLink href={spaceUrl} data-test-subj={`space-link-${space.id}`}>
+        <EuiText size="s">
+          <strong>{space.name}</strong>
+        </EuiText>
+      </EuiLink>
+    );
+  };
+
+  private renderSpaceDescription = (space: Space) => {
+    const description = space.description || '';
+
+    if (!description) {
+      return null;
+    }
+
+    const needsTruncation = description.length > 100;
+    const displayDescription = needsTruncation ? description.substr(0, 100) + '…' : description;
+
+    return (
+      <EuiTextColor color="subdued">
+        <EuiText size="s" title={description}>
+          {displayDescription}
+        </EuiText>
+      </EuiTextColor>
+    );
+  };
+
+  private renderSpaceSolution = (space: Space) => {
+    if (!space.solution) {
+      return null;
+    }
+
+    return <SpaceSolutionBadge solution={space.solution} />;
+  };
+
+  public render() {
+    const { spaces, loading } = this.props;
+    const { pageIndex, pageSize, sortField, sortDirection } = this.state;
+
+    // Sort spaces
+    const sortedSpaces = [...spaces].sort((a, b) => {
```

Comment:
Spaces is already an array
```suggestion
    const sortedSpaces = spaces.sort((a, b) => {
```